### PR TITLE
Improve readability of the README

### PR DESCRIPTION
**Pull request by evuez**
It's just small stuff but hopefully it'll make it a *little* more
pleasant to read.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,24 +74,24 @@ features = ["regexp"]
...
 The resulting code is small, and looks like the grammar you would have written with other parser approaches.
 
 This has a few advantages:
 
-- the parsers are small and easy to write
-- the parsers components are easy to reuse (if they're general enough, please add them to nom!)
-- the parsers components are easy to test separately (unit tests and property-based tests)
-- the parser combination code looks close to the grammar you would have written
-- you can build partial parsers, specific to the data you need at the moment, and ignore the rest
+- the parsers are small and easy to write;
```
Comment (Geal) [inline README.md]: this is not really necessary

File: README.md [hunk 2]
```diff
@@ -157,24 +158,24 @@ pub enum Needed {
...
+- **`length_value`**: a byte indicating the size of the following buffer
+- **`not_line_ending`**: returning as much data as possible until a line ending (\r or \n) is found
+- **`line_ending`**: matches a line ending
+- **`alpha`**: will return the longest alphabetical array from the beginning of the input
+- **`digit`**: will return the longest numerical array from the beginning of the input
+- **`alphanumeric`**: will return the longest alphanumeric array from the beginning of the input
+- **`space`**: will return the longest array containing only spaces
+- **`multispace`**: will return the longest array containing space, \r or \n
+- **`be_u8`**, **`be_u16`**, **`be_u32`**, **`be_u64`** to parse big endian unsigned integers of multiple sizes
+- **`be_i8`**, **`be_i16`**, **be_i32``**, **`be_i64`** to parse big endian signed integers of multiple sizes
```
Comment (Geal) [inline README.md]: there's a typo on be_i32

**Comment by evuez** [comment]
@Geal I fixed the `be_i32` thing and removed the semicolons, tell me if there's anything else you'd like me to change 😃

**Comment by Geal** [comment]
it's alright, thanks!

**Pull request merged**